src/project/manifest/mod.rs

```diff
@@ -338,26 +353,46 @@ impl Manifest {
         &mut self,
         dep: &rip::types::PackageName,
         platform: Option<Platform>,
+        feature_name: Option<&FeatureName>,
     ) -> miette::Result<(rip::types::PackageName, PyPiRequirement)> {
-        get_toml_target_table(&mut self.document, platform, consts::PYPI_DEPENDENCIES)?
-            .remove(dep.as_str())
-            .ok_or_else(|| {
-                let table_name = match platform {
-                    Some(platform) => {
-                        format!("target.{}.{}", platform.as_str(), consts::PYPI_DEPENDENCIES)
-                    }
-                    None => consts::PYPI_DEPENDENCIES.to_string(),
-                };
+        get_toml_target_table(
+            &mut self.document,
+            platform,
+            feature_name,
+            consts::PYPI_DEPENDENCIES,
+        )?
+        .remove(dep.as_source_str())
+        .ok_or_else(|| {
+            let table_name = match (platform, feature_name) {
```

Comment:
The function I mentioned above could also be used here!

Reply:
Changed in https://github.com/prefix-dev/pixi/pull/680/commits/5556e0b53355a9bfd6e7fd227ac22cea19063cfc